helpers/api/marketplace/management/update-allow-list.ts

```diff
@@ -0,0 +1,58 @@
+import { isAddress } from "web3-utils";
+
+import Database from "db/models";
+
+import { getAllowListColumnFromType } from "helpers/marketplace";
+import { lowerCaseCompare, lowerCaseIncludes } from "helpers/string";
+
+import { AllowListTypes } from "interfaces/enums/marketplace";
+
+import { ErrorMessages } from "server/errors/error-messages";
+import { HttpBadRequestError } from "server/errors/http-errors";
+
+export async function updateAllowListByType (address: string,
+                                             networkId: number,
+                                             type: AllowListTypes,
+                                             operation: "add" | "remove"): Promise<string[]> {
+  const isValidType = Object.values(AllowListTypes).includes(type);
+
+  if (!address || !networkId || (address && !isAddress(address as string)) || !type || !isValidType)
+    throw new HttpBadRequestError(ErrorMessages.InvalidPayload);
+
+  const listColumn = getAllowListColumnFromType(type);
+
+  const network = await Database.network.findOne({
+    attributes: [listColumn],
+    where: {
+      id: networkId,
+    }
+  });
+
+  if (!network)
+    throw new HttpBadRequestError(ErrorMessages.NoNetworkFoundOrUserAllowed);
+
+  const isPresentOnList = lowerCaseIncludes(address, network[listColumn] || []);
+
+  if (operation === "add" && isPresentOnList)
+    throw new HttpBadRequestError(ErrorMessages.NoNetworkFoundOrUserAllowed);
```

Comment:
This error is wrong. It _was_ correct when the if condition was `!network || !notOnList` but not the condition is solely `not on list` thus we need to send that error alone - `UserAlreadyOnList` and `UserNotOnList`